types/index.d.ts

```diff
@@ -46,8 +46,8 @@ declare namespace mix {
         /** Run a callback after the build has completed */
         then(callback: (stats: webpack.Stats) => void | Promise<void>): Api;
 
-        /** Run a callback after the build has completed */
-        when(callback: () => void | Promise<void>): Api;
+        /** Run a callback if a condition is true */
+        when(condition: boolean, callback: (mix: Api) => void | Promise<void>): Api;
```

Comment:
welp I totally had the types wrong for .when. 😂 

This looks 100% correct, yes! Thank you!

Reply:
*[Copied from another comment.]* Does that `| Promise<void>` look right to you? I don't think that when() would work if passed a Promise, but maybe I'm wrong.

Reply:
I'd leave it! We can always adjust the types again later if need be.